tests/lib/packaging/build-rpm.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+set -e
+
+usage() {
+    echo "Usage: $0 --pkg-dir <package> --vendor-tar-dir <vendor_tar_dir> --config-file <config_name>"
+    exit 1
+}
+
+pkg_dir=""
+vendor_tar_dir=""
+config_file=""
+
+# Parse arguments
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        --pkg-dir)
+            pkg_dir="$2"
+            shift 2
+            ;;
+        --vendor-tar-dir)
+            vendor_tar_dir="$2"
+            shift 2
+            ;;
+        --config-file)
+            config_file="$2"
+            shift 2
+            ;;
+        *)
+            echo "Unknown option: $1"
+            usage
+            ;;
+    esac
+done
+
+# Check required arguments
+if [ -z "$pkg_dir" ] || [ -z "$vendor_tar_dir" ] || [ -z "$config_file" ]; then
+    usage
+fi
+
+src_dir=/tmp/sources
+
+mkdir "$src_dir"
+
+version=$(cat "$vendor_tar_dir"/version)
+packaging_path=packaging/"$pkg_dir"
+
+sed -i -e "s/^Version:.*$/Version: $version/g" "$packaging_path/snapd.spec"
```

Comment:
It really feels like the version editing should be done in a separate job step in the caller rather than here (similarly for deb/arch builds)